ground/src/main/java/com/google/android/ground/system/GoogleApiManager.kt

```diff
@@ -52,11 +52,11 @@ constructor(
       } else if (googleApiAvailability.isUserResolvableError(status)) {
         activityStreams.withActivity {
           googleApiAvailability.showErrorDialogFragment(it, status, INSTALL_API_REQUEST_CODE) {
-            emitter.onError(Exception())
-          } // TODO: Throw appropriate Exception.
+            emitter.onError(Exception("Google play services not available"))
```

Comment:
> We should probably just pass the exception here instead of returning our own generic one. At the very least, we should propagate the underlying exception in the new Exception e -> Exception("Google play ....", e)

We emit an error when the GooglePlayServicesErrorDialog is canceled. So, there is no exception passed to the block. 

>  If possible, we should also use a more specific class than Exception.

detekt does contain a rule which prevents using a generic exception or error. We can add it later.

Reply:
Ah, ok, sounds good!